### PR TITLE
Fixes autofire, for real this time.

### PR DESCRIPTION
**Pull request by Kmc2000**
## About The Pull Request

Mirror of https://github.com/BeeStation/NSV13/pull/814. Please direct review there to avoid headache.

## Why It's Good For The Game

-Autofire code has been taken out the back and shot.  It now uses a component and is far more consistent in its behaviour, allowing you to do epic things like place guns on a table without them breaking EVERYTHING.

-Full-auto has been moved to a small focus group of guns for the time being. Admins can edit this in game by changing the full_auto var

-Autofire now uses fire_rate to determine how quickly you can shoot.

-Autofire no longer lets you magdump people (As far as I can tell through our testing)

-Autofire no longer bugs the hell out when trying to physically click on people, throw the gun etc.

## Changelog
:cl: Kmc2000 with Gun Jesus
add: Fully automatic firing for guns, except this time it isn't horribly coded and broken.
/:cl:

### CODE REVIEW

**Comment by Kmc2000** [comment]
This is currently test merged on NSV. I'll get back to you on how that goes

**Comment by Kmc2000** [comment]
We've had this running on NSV for the weekend and I've not had the usual complaints about autofire being broken. We'll be merging it on our end, recommending you testmerge this yourselves!

File: code/modules/projectiles/autofire.dm [hunk 1]
```diff
@@ -1,70 +1,131 @@
...
+			else //They're trying to disable the full auto of a gun. Remove the relevent component
+				if(autofire_component)
+					autofire_component.RemoveComponent()
+					qdel(autofire_component)
+					return ..()
+		if(NAMEOF(src, fire_rate))
+			autofire_component?.default_fire_delay = (10 / var_value)
+			return ..()
+		else
+			return ..()
```
Comment (PowerfulBacon) [inline code/modules/projectiles/autofire.dm]: ..() called twice. remove the top . = ..() since its made redundant by line 36 anyway

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Remove the double ..()

File: code/modules/projectiles/autofire.dm (hunk 1 above)
Comment (PowerfulBacon) [inline code/modules/projectiles/autofire.dm]:
```suggestion
					return
			else //They're trying to disable the full auto of a gun. Remove the relevent component
				if(autofire_component)
					autofire_component.RemoveComponent()
					qdel(autofire_component)
					return ..()
		if(NAMEOF(src, fire_rate))
			autofire_component?.default_fire_delay = (10 / var_value)
			return
```

**Review by PowerfulBacon — APPROVED** [review]

**Comment by AnjelicaChaykovsky** [comment]
hos gun is not full auto anymore because of this i guess ?

**Pull request merged**